bin/check-puppet-last-run.rb

```diff
@@ -95,7 +107,11 @@ def run
       @message += " with #{@failures} failures"
     end
 
-    if @now - @last_run > config[:crit_age] || @failures > 0
+    if @restart_failures > 0
```

Comment:
There are four possibilities:
1. Restart failure happened in last run and user has enabled option to report it - puppet report section for resources would be read and restart_failures would contain value higher then zero and this code would be triggered
2. Restart failure happened in last run and user had not enabled option to report it - this would skip checking for value in puppet report and assign value of 0 to restart_failures so code would not be triggered
3. There were no restart failures and user has enabled restart failures reporting - this would cause value of 0 to be read from puppet report and code would not be triggered
4. There were no restart failures and user had not enabled failures reporting - same thing would happen as in point 2, instead of reading value from puppet report value of 0 will be assigned and code would be not triggered.
Hence I skipped here checks for config[:report_restart_failures] as they seemed redundant.